src/GAP/iso_oscar_gap.jl

```diff
@@ -30,10 +30,42 @@ end
 #
 ################################################################################
 
+@attributes Nemo.NmodRing # TODO: port this to Nemo
+@attributes Nemo.FmpzModRing # TODO: port this to Nemo
+
+# Assume that `RO` and `RG` are residue rings of the same size
+# in Oscar and GAP, respectively.
+function _iso_oscar_gap_residue_ring_functions(RO::Union{Nemo.NmodRing, Nemo.FmpzModRing}, RG::GAP.GapObj)
+   e = GAPWrap.One(RG)
+   f(x) = GAP.Obj(lift(x))*e
+
+   finv = function(x::GAP.Obj)
+     if GAP.Globals.IsFFE(x)
+       y = GAP.Globals.Int(x)
+     elseif GAP.Globals.IsZmodnZObj(x)
+       y = GAP.Globals.ExtRepOfObj(x)
+     else
+       error("GAP object $x is not supported")
+     end
+     return y isa Int ? RO(y) : RO(fmpz(y))
```

Comment:
- meta: we have many places which convert `GapInt` to `Union{Int,fmpz}` by preserving Ints and mapping GAP large ints to fmpz. It might be nice to have a helper for that...